### PR TITLE
1156 index analysis

### LINKED ISSUE #1156: Index analysis + error information
### Summary
<table>
  <tr>
    <td><b>As a</b></td>
    <td>metadata manager</td>
  </tr>
  <tr>
    <td><b>I want to be able to</b></td>
    <td>
       see analysis and error information in a dashboard
    </td>
  </tr>
  <tr>
    <td><b>So that I can</b></td>
    <td>observe information about the metadata inventory to identify errors and potential improvements</td>
  </tr>
</table>

### Background

We already collect both `analysis` and `errors` information from each `ParsedRecord` that flows through our system, e.g. whether or not it has a title or if some value could not be parsed. This information should be indexed in Elasticsearch so it can be easily observed through Kibana. At the moment, you can only retrieve all this information by explicitly GETing a given record from the registry's REST API. We've demonstrated pushing some of this information into Elasticsearch via Kafka Connect, but we'd prefer to index it using the new indexer module so that:

- We no longer need to deploy Kafka Connect
- We can control the related Elasticsearch index mapping(s)
- You get the analysis and error information for free when you deploy the OSIM system

### Tasks
- [ ] Decide how to organize analytical and error information in Elasticsearch, e.g. should they be in separate indices?
- [ ] Add Elasticsearch index mappings for the record-level analysis and error information and apply them with `indexer`
- [ ] Index the `analysis` block of information for every incoming record
- [ ] Index any `errors` for every incoming record (and remove them from ES if there aren't any anymore)
- [ ] (optional) Expand application config / ES client logic to support sending this info to a separate ES cluster than the search info
- [ ] (optional) Index OneStop validation errors in addition to to the `analysis` block so users can discover why records are not discoverable

### PR DESCRIPTION
**Pull request by ajakz**
This PR adds support for indexing the Analysis and Errors information for both granules and collections.

Work done:
- Consolidated config into less overall locations and utilized less references to configs and ES service overall
- Made IndexingInput more self-aware for clarity and ease of use
- Added mappings for collection and granule A&E indices (note -- content of indices shouldn't be focused on too much here as that is the focus of another story #1238 and a follow-on for iterative development with the end users)
- Added extra validation for topic placement with record type

Depends on cedardevs/schemas#31 (Note -- will need to merge that PR first, then update the `ONESTOP_SCHEMAS` value in `buildSrc/src/main/kotlin/utils.kt` in this branch prior to merging)
Resolves #1156

### CODE REVIEW

File: elastic-common/src/main/java/org/cedar/onestop/elastic/common/ElasticsearchConfig.java [hunk 1]
```diff
@@ -36,7 +37,9 @@
 
   private Map<String, String> jsonMappings = new HashMap<>();
   private Map<String, Map<String, Map>> parsedMappings = new HashMap<>();
-  private Map<String, String> typesByAlias = new HashMap<>();
+  private Map<String, String> typeByAlias = new HashMap<>();
+  private Map<String, String> searchAliasByType = new HashMap<>();
+  private Map<String, String> aeAliasByType = new HashMap<>();
```
[the remark this answers is not in the document]
Comment (ajakz) [reply elastic-common/src/main/java/org/cedar/onestop/elastic/common/ElasticsearchConfig.java:42]: either way, I was just admittedly getting lazy typing all of it every time

File: indexer/src/main/java/org/cedar/onestop/indexer/util/IndexingInput.java [hunk 2]
```diff
@@ -33,19 +30,53 @@ public String getKey() {
...
+    }
+    else {
+      return new HashSet<>();
+    }
   }
 
-  public ElasticsearchConfig getEsConfig() {
-    return esConfig;
+  public Set<String> getTargetAnalysisAndErrorsIndexFields() {
+    var aeAlias = esConfig.analysisAndErrorsAliasFromType(recordType.toString());
```
[the remark this answers is not in the document]
Comment (ajakz) [reply indexer/src/main/java/org/cedar/onestop/indexer/util/IndexingInput.java:66]: either way, I was just admittedly getting lazy typing all of it every time

File: indexer/src/main/java/org/cedar/onestop/indexer/util/IndexingUtils.java [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+      }
+    }
+    return requests;
+  }
+
+  public static boolean isTombstone(ParsedRecord value) {
+    return value == null;
+  }
+
+  public static boolean isPrivate(ParsedRecord value) {
```
[the remark this answers is not in the document]
Comment (ajakz) [reply indexer/src/main/java/org/cedar/onestop/indexer/util/IndexingUtils.java:50]: there may or may not already be a story but as far as I'm aware right now there's not yet a use case to see private data in the search index, so the first story would probably be "determine the desired use case"

File: indexer/src/main/java/org/cedar/onestop/indexer/util/IndexingInput.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (StrayCat1) [reply indexer/src/main/java/org/cedar/onestop/indexer/util/IndexingInput.java:66]: changed for clarity here

**Pull request merged**